unyt/dimensions.py

```diff
@@ -277,7 +277,9 @@ def new_f(*args, **kwargs):
                 If the units do not match.
 
             """
-            for arg_name, arg_value in chain(zip(names_of_args, args), kwargs.items()):
+            for arg_name, arg_value in chain(
+                zip(names_of_args, args, strict=False), kwargs.items()
```

Comment:
this is the one place where I used `strict=False` because it makes sense in this context: when all arguments are passed as keywords, `args` is empty an d `names_of_args` is effectively the same as `list(kwargs.keys())`